.github/workflows/check_task.yml

```diff
@@ -12,8 +10,82 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  check-affected-files:
+    name: Check affected files
+    runs-on: ubuntu-latest
+    outputs:
+      affected_file: ${{ steps.check-affected-files.outputs.affected_file }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v45
+
+      - name: Check affected files
+        id: check-affected-files
+        env:
+          # get all affected files (created, modified, deleted, etc.)
+          FILES: ${{ steps.changed-files.outputs.all_changed_and_modified_files }}
+          FILES_COUNT: ${{ steps.changed-files.outputs.all_changed_and_modified_files_count }}
+        run: |
+          for file in "$FILES"; do
+            echo "Info: $file was changed"
+          done
+          if [ "$FILES_COUNT" -ne 1 ]; then
+            echo "Error: More than one file affected ($FILES_COUNT)"
+            exit 1
+          fi
+          if [[ "$FILES" =~ ^contributions\/(executable-tutorial|feedback|open-source)\/[^\/]+\/README\.md$ ]]; then
+            echo "Info: Matches async task"
+          elif [[ "$FILES" =~ ^contributions\/(demo|presentation|scientific-paper)\/week[2-7]\/[^\/]+\/README\.md$ ]]; then
+            echo "Info: Matches sync task"
+          else
+            echo "Error: File in wrong directory ($FILES)"
+            exit 1
+          fi
+
+          echo "affected_file=$FILES" >> "$GITHUB_OUTPUT"
+
+  check-pr-description:
+    name: Check whether Pull Request description matches affected file
+    needs: check-affected-files
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
This should trigger on description edit as well.

Reply:
Fixed in bd6321d30fb564ffed6b567ef5e98e8ea01846c1